src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkitemOrchestrator.cs

```diff
@@ -21,5 +23,13 @@ public interface IWorkitemOrchestrator
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A task that represents the asynchronous orchestration of the adding operation.</returns>
         Task AddWorkitemAsync(DicomDataset dataset, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Asynchronously orchestrate the searching of a UPS-RS workitem
+        /// </summary>
+        /// <param name="parameters"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<QueryWorkitemResourceResponse> QueryAsync(QueryParameters parameters, CancellationToken cancellationToken = default);
```

Comment:
Between the class comment of use in Store orchestration, I am not sure if you need QuerAsync here and WorkItemService

Reply:
We are just keeping in Orchestrator to utilize the already initialized arguments in the orchestrator. May be we can call the orchestrator directly from the handler.

Reply:
We've generally tried to follow the pattern of: authz and content negotiation in the handler, core business logic in the service, and SQL + blob persistence coordination (two-step commit, compensating commit) in the orchestration. In this particular case, since we are handling multiple routes in the service as opposed to QIDO which only handles querying, it seemed like a cleaner implementation to have the orchestrator handle combining data from SQL and blob instead of having both classes work with both persistence providers.